src/golang/lib/collections/workflow/postgres.go

```diff
@@ -26,21 +26,28 @@ func (r *postgresReaderImpl) GetWorkflowsWithLatestRunResult(
 	ctx context.Context,
 	organizationId string,
 	db database.Database,
-) ([]latestWorkflowResponse, error) {
+) ([]LatestWorkflowResponse, error) {
 	// Get workflow metadata (id, name, description, creation time, last run time, and last run status)
 	// for all workflows whose `organization_id` is `organizationId` ordered by when the workflow was created.
 	// Get the last run DAG by getting the max created_at timestamp for all workflow DAGs associated with each
 	// workflow in the organization.
-	query := `SELECT * FROM (SELECT DISTINCT ON (wf.id) wf.id AS id, wf.name AS name, 
-		 wf.description AS description, wf.created_at AS created_at, 
-		 wfdr.created_at AS last_run_at, wfdr.status as status 
-		 FROM workflow AS wf, app_user, workflow_dag AS wfd, workflow_dag_result AS wfdr 
-		 WHERE app_user.organization_id = $1 
-		 AND wf.user_id = app_user.id AND wfd.workflow_id = wf.id AND wfdr.workflow_dag_id = wfd.id 
-		 ORDER BY wf.id, wfdr.created_at DESC) AS temp
-		 ORDER BY created_at DESC;`
-
-	var latestWorkflowResponse []latestWorkflowResponse
+	query := `
+	SELECT * FROM (
+		SELECT DISTINCT ON 
+			(wf.id) wf.id AS id, wf.name AS name, 
+		 	wf.description AS description, wf.created_at AS created_at, 
+		 	wfdr.created_at AS last_run_at, wfdr.status as status 
+		FROM 
+			workflow AS wf 
+			INNER JOIN app_user ON wf.user_id = app_user.id
+			INNER JOIN workflow_dag AS wfd ON wf.id = wfd.workflow_id
+			LEFT JOIN workflow_dag_result AS wfdr ON wfd.id = wfdr.workflow_dag_id
```

Comment:
The changes here are making all of the `INNER JOIN`s explicit and replacing the final `INNER JOIN` with a `LEFT JOIN`.

Reply:
Can you explain the left join here a little more? I've been staring at it for a while but want to make sure :P 

My understanding is that a left join takes all the records of the left table (which are `workflow` records), and any matching records on the right table. Does this mean that at least one record is returned for each workflow, but there can be workflows with no workflow dag results, meaning the `last_run_at` and `status` fields of such workflows are empty?

If thats the case, we can add a comment noting that potential nil-ness? I'd rather our SQL queries be really well documented than not since these take a lot of brainpower to read.